### ISSUE TITLE
Filter of complex queries bug

### ISSUE THREAD

**Issue by plotn** [dbeaver/dbeaver#8855]
#### System information: 
- Operating system (distribution) and version: Win10
- DBeaver version: 7.0.5

#### Connection specification:
- Database name and version: PostgrSQL 12

.. as shown
![filter_error](https://user-images.githubusercontent.com/11477081/83390121-f9b82300-a3f9-11ea-8abb-69d76d7e9691.png)

**Comment by kseniiaguzeeva** [comment]
Thank you for the report. Could you please provide with a complex query example?

**Comment by plotn** [comment]
select 
  l2.lnk_id l2_lnk_id,
  coalesce(isis.lnk_id, isis_direct.lnk_id) isis_lnk_id,
  l2.gr_rep_link_type link_type,
  coalesce(isis.gr_code_a, isis_direct.gr_code_a, l2.gr_a) gr_code_a,
  coalesce(isis.gr_code_z, isis_direct.gr_code_z, l2.gr_z) gr_code_z,
  coalesce(isis.gr_code_a, isis_direct.gr_code_a, l2.gr_a)||' - '||
      coalesce(isis.gr_code_z, isis_direct.gr_code_z, l2.gr_z) gr_direction,
  case 
    when l2.gr_rep_link_type = 'CR_RGR' then
      coalesce(isis.gr_code_a, isis_direct.gr_code_a, l2.gr_a)||' - '||
      coalesce(isis.gr_code_z, isis_direct.gr_code_z, l2.gr_z) 
    else  
      sprt_util.get_nd_name_by_router_name(
        coalesce(isis.teoco_ne_a, isis_direct.teoco_ne_a, l2.teoco_ne_a) 
      )||' - '||
      sprt_util.get_nd_name_by_router_name(
        coalesce(isis.teoco_ne_z, isis_direct.teoco_ne_z, l2.teoco_ne_z) 
      )
  end direction,  
  gr_a.gr_region || ' - ' || gr_z.gr_region direction_text,              
  trim('TEOKO'||case when coalesce(l2.sc_id_a, l2.sc_id_z) is not null then ' + СУРМС' else '' end||
    case when coalesce(o_id_a, o_id_z) is not null then ' + СУРТС' else '' end) d_source,
  case 
    when isis.lnk_id is not null then 'ISIS-LAG-L2Link'
    when isis_direct.lnk_id is not null then 'ISIS-L2link'
    else 'L2link'
  end as teoco_source,  
  true is_active,  
  coalesce(isis.crcr_order, isis_direct.crcr_order) crcr_order,
  coalesce(isis.lnk_bandwidth_bits, isis_direct.lnk_bandwidth_bits, l2.lnk_bandwidth_bits) lnk_bandwidth_bits,
  coalesce(l2.if_bandwidth_bits_surms_a, l2.if_bandwidth_bits_surms_z) lnk_bandwidth_bits_surms,
  gt.gt_type,
  l2.distance l2_distance,
  l2.is_transport,
  l2.channel_type 
--  l2.*
from dwh.teoco_l2_surms_conn_with_distance l2
left join (
       select isis1.*, 
         lags_a.lag_identifier lags_a_lag_identifier,
         lags_a.port_identifier lags_a_port_identifier, 
         lags_z.lag_identifier lags_z_lag_identifier,
         lags_z.port_identifier lags_z_port_identifier
       from dwh.v_links_data_all_teoco_isis_and_l2 isis1     
       join dwh.teoco_lag_dwh lags_a ON lags_a.router_name = isis1.teoco_ne_a 
         and lags_a.lag_identifier = isis1.teoco_port_a AND lags_a.tl_del_dt IS null
       join dwh.teoco_lag_dwh lags_z ON lags_z.router_name = isis1.teoco_ne_z
         and lags_z.lag_identifier = isis1.teoco_port_z AND lags_z.tl_del_dt IS null        
       where isis1.lnk_topo_object_type::text = 'ISIS' 
) isis 
on isis.rep_link_type = l2.gr_rep_link_type 
   and isis.teoco_ne_a = l2.teoco_ne_a and isis.teoco_ne_z = l2.teoco_ne_z
   and l2.teoco_port_a = isis.lags_a_port_identifier
   and l2.teoco_port_z = isis.lags_z_port_identifier
left join dwh.v_links_data_all_teoco_isis_and_l2 isis_direct
  on isis_direct.lnk_topo_object_type::text = 'ISIS' and isis_direct.rep_link_type = l2.gr_rep_link_type and 
    isis_direct.teoco_ne_a = l2.teoco_ne_a and isis_direct.teoco_ne_z = l2.teoco_ne_z and
    isis_direct.teoco_port_a = l2.teoco_port_a and isis_direct.teoco_port_z = l2.teoco_port_z
left join sprt_ref.ref_groups gr_a on gr_a.gr_code = coalesce(isis.gr_code_a, isis_direct.gr_code_a, l2.gr_a)
  and current_timestamp between gr_a.gr_start_date and gr_a.gr_end_date 
left join sprt_ref.ref_groups gr_z on gr_z.gr_code = coalesce(isis.gr_code_z, isis_direct.gr_code_z, l2.gr_z)
  and current_timestamp between gr_z.gr_start_date and gr_z.gr_end_date   
left join sprt_ref.crcr_group_type gt on 
	  trim(replace(gt.gt_group, 'М', 'M')) = 
	    trim(replace(coalesce(isis.gr_code_a, isis_direct.gr_code_a, l2.gr_a)||' - '||
        coalesce(isis.gr_code_z, isis_direct.gr_code_z, l2.gr_z), 'М', 'M'))
where gr_rep_link_type in ('CR_CR', 'CR_RGR');

**Comment by kseniiaguzeeva** [comment]
Sorry for the long response. I cannot execute the query. Error appears "Relation "dwh.teoco_l2_surms_conn_with_distance" does not exist"
What's can be wrong?

**Comment by plotn** [comment]
;))) I guess you do not have tables, that I mentioned. The query given just as an example. I thought that you can find this bug without real tables and data (see my message "maybe we should add an alias").

**Comment by kseniiaguzeeva** [comment]
You are right, if you change SQL expression and add aliases, it will solve the issue.
From our side an enhancement to work with non-unique columns will be added in the future.

**Comment by plotn** [comment]
It is not my fault but yours - the query runs correct. Then you slightly modify it on your side to include filter from UI, but your resulting query becomes broken. Thus, you should enchance your "sql transformer for filters".

Как мне кажется....
